crates/ruma-signatures/src/functions.rs

```diff
@@ -736,29 +736,23 @@ pub fn redact_in_place(
     event: &mut CanonicalJsonObject,
     version: &RoomVersionId,
 ) -> Result<(), Error> {
-    let event_type_value = match event.get("type") {
-        Some(event_type_value) => event_type_value,
+    // Get the content keys here instead of the event type, because we cant teach rust that this is
+    // a disjoint borrow.
+    let allowed_content_keys: &[&str] = match event.get("type") {
+        Some(CanonicalJsonValue::String(event_type)) => {
+            allowed_content_keys_for(event_type.as_ref(), version)
```

Comment:
I think the `.as_ref()` isn't needed.

```suggestion
            allowed_content_keys_for(event_type, version)
```